dali/pipeline/data/tensor_vector.h

```diff
@@ -78,72 +88,130 @@ class DLL_PUBLIC TensorVector {
    */
   void set_order(AccessOrder order, bool synchronize = true);
 
-  Tensor<Backend> &operator[](size_t pos) {
-    return *(tensors_[pos]);
-  }
-
-  const Tensor<Backend> &operator[](size_t pos) const {
-    return *(tensors_[pos]);
-  }
-
-  auto tensor_handle(size_t pos) {
-    return tensors_[pos];
-  }
-
-  auto tensor_handle(size_t pos) const {
-    return tensors_[pos];
+  SampleView<Backend> operator[](size_t pos) {
+    return {tensors_[pos]->raw_mutable_data(), tensors_[pos]->shape(), tensors_[pos]->type()};
   }
 
-  auto begin() noexcept {
-    return tensors_.begin();
-  }
-
-  auto begin() const noexcept {
-    return tensors_.begin();
-  }
-
-  auto cbegin() const noexcept {
-    return tensors_.cbegin();
-  }
-
-  auto end() noexcept {
-    return tensors_.end();
-  }
-
-  auto end() const noexcept {
-    return tensors_.end();
-  }
-
-  auto cend() const noexcept {
-    return tensors_.cend();
+  ConstSampleView<Backend> operator[](size_t pos) const {
+    return {tensors_[pos]->raw_data(), tensors_[pos]->shape(), tensors_[pos]->type()};
   }
 
   size_t num_samples() const noexcept {
     return curr_tensors_size_;
   }
 
+  void set_sample_dim(int sample_dim);
+
   int sample_dim() const {
-    return IsContiguous() ? tl_->sample_dim() : num_samples() ? tensors_[0]->shape().size() : 0;
+    return sample_dim_;
   }
 
   size_t nbytes() const noexcept;
 
   size_t capacity() const noexcept;
 
+  /**
+   * @brief Returns the size in bytes of the underlying data chunks
+   * TODO(klecki): Temporary API to be reworked, do not use.
+   */
+  std::vector<size_t> _chunks_nbytes() const noexcept;
```

Comment:
```suggestion
  std::vector<size_t> _chunks_nbytes() const;
```
After all, a `vector` is dynamically allocated...

Reply:
done